OrbitVulkanLayer/QueueManager.cpp

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "QueueManager.h"
+
+#include "OrbitBase/Logging.h"
+
+namespace orbit_vulkan_layer {
+
+void QueueManager::TrackQueue(const VkQueue& queue, const VkDevice& device) {
+  absl::WriterMutexLock lock(&mutex_);
+  CHECK(!queue_to_device_.contains(queue) || queue_to_device_.at(queue) == device);
```

Comment:
Have you considered to fail gracefully here? Maybe return a ErrorMessageOr<void>? I'm not sure if that's a good use-case for a CHECK here.

Reply:
In the layer we won't be able to handle errors anyways, thus I'd rather keep my expectations on the calls explicit.